packages/autorest.typescript/test/integration/generated/nameChecker/src/models/parameters.ts

```diff
@@ -330,8 +330,8 @@ export const selectedFields: OperationQueryParameter = {
   collectionFormat: "CSV",
 };
 
-export const search$DONotNormalize$Text: OperationQueryParameter = {
-  parameterPath: "search$DONotNormalize$Text",
+export const searchDONOTNormalizeText: OperationQueryParameter = {
```

Comment:
$ is removed.

Reply:
For this parameter:
1. The client name is `"x-ms-client-name": "SearchText"` and setting 
```
override:
      text: $DO_NOT_NORMALIZE$_text
```
2. so in m4 it is replaced by `Search$DO_NOT_NORMALIZE$_text`
3. HLC normalized as `search$DONotNormalize$Text`
4. if we switch to new norm algo it would become `searchDONOTNormalizeText`

Reply:
why NOT becomes Not in hlc?